### PR TITLE
sacad: init at 2.3.4

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change
add sacad

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114618` at 602d43fc run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/520df5c121baac5813b3179dfb965a7d">python37Packages.web_cache</a></li></li>
    <li><a href="https://gist.github.com/d442c0a2f98e93dfe37f1d2d73c558cd">python38Packages.web_cache</a></li></li>
    <li><a href="https://gist.github.com/816853ada174a8c9652773cb74dedae9">python39Packages.web_cache</a></li></li>
    <li><a href="https://gist.github.com/1a45baeac946ec30c88098f18c46ed82">sacad</a></li></li>
  </ul>
</details>

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -8372,6 +8372,8 @@ in {
 
   web = callPackage ../development/python-modules/web { };
 
+  web_cache = callPackage ../development/python-modules/web_cache { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:8375]:
```suggestion
  web-cache = callPackage ../development/python-modules/web-cache { };
```

File: pkgs/tools/misc/sacad/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    appdirs
+    bitarray
+    cssselect
+    fake-useragent
+    lxml
+    mutagen
+    pillow
+    tqdm
+    unidecode
+    web_cache
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sacad/default.nix:28]:
```suggestion
    web-cache
```

File: pkgs/tools/misc/sacad/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  pname = "sacad";
+  version = "2.3.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1qv2mrz6vy2sl7zhrj9vw016pjd7hmjr2ls0w8bbv1hgrddicn9r";
+  };
+
+  # tests require internet connection
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sacad/default.nix:15]:
```suggestion
```

File: pkgs/tools/misc/sacad/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    lxml
+    mutagen
+    pillow
+    tqdm
+    unidecode
+    web_cache
+    jpegoptim
+    optipng
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sacad/default.nix:32]:
```suggestion

  # tests require internet connection
  doCheck = false;

```

File: pkgs/tools/misc/sacad/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
+{ lib, python3Packages, jpegoptim, optipng }:
+
+with python3Packages;
+
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sacad/default.nix:4]:
```suggestion
```

File: pkgs/tools/misc/sacad/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+buildPythonApplication rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sacad/default.nix:5]:
```suggestion
python3Packages.buildPythonApplication rec {
```

File: pkgs/tools/misc/sacad/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1qv2mrz6vy2sl7zhrj9vw016pjd7hmjr2ls0w8bbv1hgrddicn9r";
+  };
+
+  # tests require internet connection
+  doCheck = false;
+
+  propagatedBuildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sacad/default.nix:17]:
```suggestion
  propagatedBuildInputs = with python3Packages; [
```

File: pkgs/development/python-modules/web_cache/default.nix [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+buildPythonPackage rec {
+  pname = "web_cache";
+  version = "1.1.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1d8f1s3i0s3h1jqvjq6cp639hhbbpxvyq7cf9dwzrvvvr0s0m8fm";
+  };
+
+  disabled = !isPy3k;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/web_cache/default.nix:12]:
```suggestion
```

File: pkgs/development/python-modules/web_cache/default.nix [hunk 9]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildPythonPackage, fetchPypi, isPy3k }:
+
+buildPythonPackage rec {
+  pname = "web_cache";
+  version = "1.1.0";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/web_cache/default.nix:5]:
```suggestion
  version = "1.1.0";
  disabled = !isPy3k;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/web-cache/default.nix [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "web-cache";
+  version = "1.1.0";
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1d8f1s3i0s3h1jqvjq6cp639hhbbpxvyq7cf9dwzrvvvr0s0m8fm";
+  };
+
+  # web_cache doesn't have tests
```
Comment (fabaff) [inline pkgs/development/python-modules/web-cache/default.nix:13]: The upstream repo contains [tests](https://github.com/desbma/web_cache/tree/master/tests).

File: pkgs/development/python-modules/web-cache/default.nix [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Simple Python key-value storage backed up by sqlite3 database";
+    homepage = "https://github.com/desbma/web_cache";
+    license = licenses.lgpl21Only;
```
Comment (fabaff) [inline pkgs/development/python-modules/web-cache/default.nix:19]: The licensing is not clear at first glance.

- `setup.py` says [GNU Lesser General Public License v2 or later (LGPLv2+)](https://github.com/desbma/web_cache/blob/master/setup.py#L33)
- Source code entry says [__license__ = "LGPLv2"](https://github.com/desbma/web_cache/blob/master/web_cache/__init__.py#L5)
- License file says [GNU LESSER GENERAL PUBLIC LICENSE Version 2.1](https://github.com/desbma/web_cache/blob/master/LICENSE)
- README says [LGPLv2](https://github.com/desbma/web_cache#license)
- GitHub says `LGPL-2.1 License` (that comes from the license file itself)

For me it looks like that it should be:

```suggestion
    license = licenses.lgpl2Plus;
```

To be sure, ask the upstream project what the actual license is.

File: pkgs/development/python-modules/web-cache/default.nix (hunk 10 above)
Comment (fortuneteller2k) [reply pkgs/development/python-modules/web-cache/default.nix]: Whoops, should have made it clearer that the archive `fetchPypi` fetches doesn't come with tests.

File: pkgs/tools/misc/sacad/default.nix [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    pillow
+    tqdm
+    unidecode
+    web-cache
+    jpegoptim
+    optipng
+  ];
+
+  # tests require internet connection
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sacad/default.nix:29]:
```suggestion
  doCheck = false;
  
  pythonImportsCheck = [ "sacad" ];
```

File: pkgs/development/python-modules/web-cache/default.nix [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit version;
+    pname = "web_cache";
+    sha256 = "1d8f1s3i0s3h1jqvjq6cp639hhbbpxvyq7cf9dwzrvvvr0s0m8fm";
+  };
+
+  # No tests in downloaded archive
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/web-cache/default.nix:15]:
```suggestion
  doCheck = false;
  
  pythonImportsCheck = [ "web_cache" ];
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114618` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>python37Packages.web-cache</li>
<li>python38Packages.web-cache</li>
<li>python39Packages.web-cache</li>
<li>sacad</li>
</ul>
</details>

**Pull request merged**